spec/rspec/matchers/built_in/contain_exactly_spec.rb

```diff
@@ -184,22 +184,74 @@ def array.send; :sent; end
 MESSAGE
   end
 
-  def timeout_if_not_debugging(time)
-    in_sub_process_if_possible do
-      require 'timeout'
-      return yield if defined?(::Debugger)
-      Timeout.timeout(time) { yield }
+  context "speed checks" do
+    def timeout_if_not_debugging(time)
+      in_sub_process_if_possible do
+        require 'timeout'
+        return yield if defined?(::Debugger)
+        Timeout.timeout(time) { yield }
+      end
     end
-  end
 
-  it 'fails a match of 11 items with duplicates in a reasonable amount of time' do
-    timeout_if_not_debugging(0.1) do
-      expected = [0, 1, 1,    3, 3, 3,    4, 4,    8, 8, 9   ]
-      actual   = [   1,    2, 3, 3, 3, 3,       7, 8, 8, 9, 9]
+    shared_examples "succeeds fast" do
+      it do
+        timeout_if_not_debugging(max_runtime) do
+          subject
+        end
+      end
+    end
 
-      expect {
-        expect(actual).to contain_exactly(*expected)
-      }.to fail_including("the missing elements were:      [0, 1, 4, 4]")
+    shared_examples "fails fast" do |failure_msg|
+      it do
+        timeout_if_not_debugging(max_runtime) do
+          expect {
+            subject
+          }.to fail_with(/#{Regexp.quote(failure_msg)}/)
+        end
+      end
+    end
+
+    let(:max_runtime) { 2 }
+    let(:actual) { Array.new(10_000) { rand(10) } }
```

Comment:
Suggestion: 1000 elements should be enough to see if the algorithm update works well. Then `max_runtime` can be dialed down as well, maybe back to 0.5.

Reply:
Ah sure thing, just updated to 1K elements and 0.5s timeout 👍